src/coreclr/src/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.CoreCLR.cs

```diff
@@ -44,7 +41,7 @@ internal partial class GenericArraySortHelper<T>
 
     internal interface IArraySortHelper<TKey, TValue>
     {
-        void Sort(Span<TKey> keys, Span<TValue> values, IComparer<TKey>? comparer);
+        void Sort<TComparer>(Span<TKey> keys, Span<TValue> values, TComparer comparer) where TComparer : IComparer<TKey>?;
```

Comment:
I think it would be better for TComparer to be on IArraySortHelper. Generic virtual methods are full of (performance) problems.

Reply:
@jkotas right forgot about that. I started out with `ArraySortHelper<TKey, TValue, TComparer>` but that has implications downstream with consumers and interfaces, so tried to avoid it. I'll try to change that later then.

Reply:
@jkotas fixed, PTAL.